File: csharp/ProductQuantity.cs
namespace supermarket
{
    public class ProductQuantity
    {
        private Product product;
        private double quantity;

        public ProductQuantity(Product product, double weight)
        {
            this.product = product;
            this.quantity = weight;
        }

        public Product getProduct()
        {
            return product;
        }

        public double getQuantity()
        {
            return quantity;
        }
    }

}

File: php/src/Model/ProductQuantity.php
<?php

declare(strict_types=1);

namespace Supermarket\Model;

class ProductQuantity
{
    public function __construct(
        private Product $product,
        private float $quantity
    ) {
    }

    public function getProduct(): Product
    {
        return $this->product;
    }

    public function getQuantity(): float
    {
        return $this->quantity;
    }
}
